tsdb/index.go

```diff
@@ -1524,9 +1524,9 @@ func (is IndexSet) MeasurementNamesByPredicate(auth query.FineAuthorizer, expr i
 
 	// Return filtered list if expression exists.
 	if expr != nil {
-		itr, err := is.measurementNamesByPredicate(auth, expr)
-		if err != nil {
-			return nil, err
+		itr, returnErr := is.measurementNamesByPredicate(auth, expr)
```

Comment:
FYI usually I would rename the returned error to a unique name rather than the shadowing one, but this works.